### DISCUSSION TITLE
How To Stream in 1080p with 60 FPS ?

### DISCUSSION THREAD

**Discussion by cortnum** [HaishinKit/HaishinKit.swift#1229] [Q&A]
Hey, I was trying to get my stream working in 1080p @ 60 FPS, but I'm not sure how to set up the configurations..

I have followed the config in the documentation and am getting it to work with 720p @ 60 FPS, but as soon as I change to 1080p or 60 FPS, it doesn't stream anymore. I will paste my code below..

Any help would be appreciated 

        self.rtmpStream = RTMPStream(connection: self.rtmpConnection)
        
        //Make sure its what we set as default in our override var preferredInterfaceOrientationForPresentation
        self.rtmpStream?.videoOrientation = .landscapeRight //NOTE: This causes crash sometimes....
        
        self.updateStreamOrientation()
        
        // Specifies the video codec settings.
        self.rtmpStream!.videoSettings = [
          .width: 1920, // video output width
          .height: 1080, // video output height
          .bitrate: 5000 * 1000, // video output bitrate
          .profileLevel: kVTProfileLevel_H264_Baseline_3_1, // H264 Profile require "import VideoToolbox"
          .maxKeyFrameIntervalDuration: 1, // key frame / sec
        ]
        
        self.rtmpStream!.attachAudio(AVCaptureDevice.default(for: AVMediaType.audio)) { error in
             debugPrint("#live error: \(error)")
        }
        
        self.rtmpStream!.attachCamera(AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: self.currentCameraPosition)) { error in
            debugPrint("#live error: \(error)")
        }
        
        self.rtmpStream!.videoCapture(for: 0)?.isVideoMirrored = self.currentCameraPosition == .front
        self.rtmpStream!.videoCapture(for: 0)?.preferredVideoStabilizationMode = .off

        self.cameraPreviewHKView.videoGravity = AVLayerVideoGravity.resizeAspectFill
        self.cameraPreviewHKView.attachStream(self.rtmpStream)
        
        
        self.rtmpStream?.frameRate = 60
        self.rtmpStream?.sessionPreset = .hd1920x1080

        self.rtmpStream?.audioSettings = [
            .bitrate: 128 * 1000,
        ]

**Comment by cortnum** [comment]
So I figured out that to bump the quality to 1080p, I need to use a higher profileLevel, I opted for kVTProfileLevel_H264_High_4_2.

Still can't quite get it to stream in 60 fps.

I am also only getting around 700 kb/s of outgoing bitrate, which seems very low compared to what I get from OBS which on average is 5000 kb/s. Even accounting for the lower 30 fps vs 60 fps, 700 still seems very low

**Reply by shogo4405** [reply]
Please perform the initialization of sessionPreset first, and then proceed with setting the frameRate.
```
  self.rtmpStream?.sessionPreset = .hd1920x1080
  self.rtmpStream?.frameRate = 60
```

**Reply by al3x88** [reply]
@shogo4405 this property is not available anymore in the last version?

**Reply by shogo4405** [reply]
Please see this [api document](
https://docs.haishinkit.com/swift/2.0.1/Actors/MediaMixer.html#/s:10HaishinKit10MediaMixerC16setSessionPresetyySo09AVCapturefG0aF).
